### DISCUSSION TITLE
No matching distribution found for stt

### DISCUSSION THREAD

**Discussion by lyjzsyzlt** [coqui-ai/STT#1804] [Q&A]
can't install stt under tutorial of https://stt.readthedocs.io/en/latest/

**Comment by reuben** [comment]
Can you post the entire log of the installation? And make sure you're using an up-to-date version of `pip` when installing.

**Reply by lyjzsyzlt** [reply]
![image](https://user-images.githubusercontent.com/30644064/111975930-144fba00-8b3c-11eb-8ea0-d6458c61ca1e.png)
My python version is 3.6.10

**Reply by reuben** [reply]
Please copy and paste the text. No screenshots.

**Reply by lyjzsyzlt** [reply]
Sorry.  
$$:python3 -m pip install -U pip && python3 -m pip install stt

log:
Collecting pip
  Downloading pip-21.0.1-py3-none-any.whl (1.5 MB)
     |████████████████████████████████| 1.5 MB 33 kB/s
Installing collected packages: pip
  Attempting uninstall: pip
    Found existing installation: pip 20.0.2
    Uninstalling pip-20.0.2:
      Successfully uninstalled pip-20.0.2
Successfully installed pip-21.0.1
ERROR: Could not find a version that satisfies the requirement stt
ERROR: No matching distribution found for stt

**Reply by reuben** [reply]
The text looks different from the screenshot? Try calling it with more verbosity using `python3 -m pip -vvv install stt`

**Reply by lyjzsyzlt** [reply]
$$:python3 -m pip -vvv install stt

Using pip 21.0.1 from /home/lvyongjie/work/venv-stt/lib/python3.6/site-packages/pip (python 3.6)
Non-user install because user site-packages disabled
Created temporary directory: /tmp/pip-ephem-wheel-cache-6idydgp6
Created temporary directory: /tmp/pip-req-tracker-1q78y9ns
Initialized build tracking at /tmp/pip-req-tracker-1q78y9ns
Created build tracker: /tmp/pip-req-tracker-1q78y9ns
Entered build tracker: /tmp/pip-req-tracker-1q78y9ns
Created temporary directory: /tmp/pip-install-jy7f58f3
1 location(s) to search for versions of stt:
* https://pypi.org/simple/stt/
Fetching project page and analyzing links: https://pypi.org/simple/stt/
Getting page https://pypi.org/simple/stt/
Found index url https://pypi.org/simple
Looking up "https://pypi.org/simple/stt/" in the cache
Request header has "max_age" as 0, cache bypassed
Starting new HTTPS connection (1): pypi.org:443
https://pypi.org:443 "GET /simple/stt/ HTTP/1.1" 304 0
  Skipping link: none of the wheel's tags match: cp35-cp35m-macosx_10_14_x86_64: https://files.pythonhosted.org/packages/2b/ad/e10cb893ac20212a59f92c0eaa8731881eca4ce20aa4665c3419b8a276c7/STT-0.10.0a4-cp35-cp35m-macosx_10_14_x86_64.whl#sha256=0c14ade5a31f125a992162f07dfe6fbb8ac8117ab9dd14978a882558ad5ace13 (from https://pypi.org/simple/stt/)
  Skipping link: none of the wheel's tags match: cp36-cp36m-macosx_10_14_x86_64: https://files.pythonhosted.org/packages/e1/fc/e06e0829d26ca2c812d6d6860dc999ffa296534012dcaca077289ea7859c/STT-0.10.0a4-cp36-cp36m-macosx_10_14_x86_64.whl#sha256=1b1614e93efca970fa73cfc3489c72676b116ffa62c2d45da2100f8471408354 (from https://pypi.org/simple/stt/)
  Skipping link: none of the wheel's tags match: cp36-cp36m-manylinux_2_24_x86_64: https://files.pythonhosted.org/packages/1a/d8/1b80ef632896394868f7e151a5c46e95c1cf8d12db377d8ba18da33cd07a/STT-0.10.0a4-cp36-cp36m-manylinux_2_24_x86_64.whl#sha256=5f91e7c5c0a161f5e890cbf76e547e084025ef832f12916a21437f4b98965bb7 (from https://pypi.org/simple/stt/)
  Skipping link: none of the wheel's tags match: cp37-cp37m-macosx_10_14_x86_64: https://files.pythonhosted.org/packages/0d/f5/8c1245a6e625312b528fe78894ce57e612a9705394b8eaa996d41c8d6dfb/STT-0.10.0a4-cp37-cp37m-macosx_10_14_x86_64.whl#sha256=cea88d277e7f512a51323707b3f8b8ab620ee55086ddf62d6e0491b2d93b67ab (from https://pypi.org/simple/stt/)
  Skipping link: none of the wheel's tags match: cp37-cp37m-manylinux_2_24_x86_64: https://files.pythonhosted.org/packages/15/97/525327fcd252e4e35f3efa26bbe16e7bf8786e9a5de725a564f531708d4b/STT-0.10.0a4-cp37-cp37m-manylinux_2_24_x86_64.whl#sha256=32bf8dc86a64f3918db182df88a684694ea1dbabd1f90886cbb7ebda251e45a0 (from https://pypi.org/simple/stt/)
  Skipping link: none of the wheel's tags match: cp38-cp38-macosx_10_14_x86_64: https://files.pythonhosted.org/packages/75/89/a283e91debf4ddcd9511ab00cb0fd9bfbead76f9e1f86a5855ce8df3a0a8/STT-0.10.0a4-cp38-cp38-macosx_10_14_x86_64.whl#sha256=47cc127bfd374e063b8a525b2de8521c2247fbf96488636fe6f764cf28c25327 (from https://pypi.org/simple/stt/)
  Skipping link: none of the wheel's tags match: cp38-cp38-manylinux_2_24_x86_64: https://files.pythonhosted.org/packages/71/02/0944668eaace71af9c37494fbc1346ce0bea8198b2d2c448d5e92ded4b62/STT-0.10.0a4-cp38-cp38-manylinux_2_24_x86_64.whl#sha256=a69dacd0de9bd1c5beaa16d40c6d592fa2fec5f949afecc488d9ea3eb16ef89a (from https://pypi.org/simple/stt/)
  Skipping link: none of the wheel's tags match: cp39-cp39-macosx_10_14_x86_64: https://files.pythonhosted.org/packages/0c/04/a3cdebc9aced6f07ce2b111496aaaa7ffe81cb9772ca07b2c2046f3c1eb1/STT-0.10.0a4-cp39-cp39-macosx_10_14_x86_64.whl#sha256=8d648aceb48b3d99b16c8d856305cbd4acc075d8c2fbb3fc85dec17e6a6e7426 (from https://pypi.org/simple/stt/)
  Skipping link: none of the wheel's tags match: cp39-cp39-manylinux_2_24_x86_64: https://files.pythonhosted.org/packages/f5/c6/8243e5509bd25a59b6c6aa9aac033e115bdd64ed763a04079a1d8391093a/STT-0.10.0a4-cp39-cp39-manylinux_2_24_x86_64.whl#sha256=0bfff32e80b88dcddfe64e6e4708ac3777f5676d5e331c947ddab2cdd371174d (from https://pypi.org/simple/stt/)
Given no hashes to check 0 links for project 'stt': discarding no candidates
ERROR: Could not find a version that satisfies the requirement stt
ERROR: No matching distribution found for stt
Exception information:
Traceback (most recent call last):
  File "/home/lvyongjie/work/venv-stt/lib/python3.6/site-packages/pip/_vendor/resolvelib/resolvers.py", line 171, in _merge_into_criterion
    crit = self.state.criteria[name]
KeyError: 'stt'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/lvyongjie/work/venv-stt/lib/python3.6/site-packages/pip/_vendor/resolvelib/resolvers.py", line 318, in resolve
    name, crit = self._merge_into_criterion(r, parent=None)
  File "/home/lvyongjie/work/venv-stt/lib/python3.6/site-packages/pip/_vendor/resolvelib/resolvers.py", line 173, in _merge_into_criterion
    crit = Criterion.from_requirement(self._p, requirement, parent)
  File "/home/lvyongjie/work/venv-stt/lib/python3.6/site-packages/pip/_vendor/resolvelib/resolvers.py", line 83, in from_requirement
    raise RequirementsConflicted(criterion)
pip._vendor.resolvelib.resolvers.RequirementsConflicted: Requirements conflict: SpecifierRequirement('stt')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/lvyongjie/work/venv-stt/lib/python3.6/site-packages/pip/_internal/resolution/resolvelib/resolver.py", line 122, in resolve
    requirements, max_rounds=try_to_avoid_resolution_too_deep,
  File "/home/lvyongjie/work/venv-stt/lib/python3.6/site-packages/pip/_vendor/resolvelib/resolvers.py", line 453, in resolve
    state = resolution.resolve(requirements, max_rounds=max_rounds)
  File "/home/lvyongjie/work/venv-stt/lib/python3.6/site-packages/pip/_vendor/resolvelib/resolvers.py", line 320, in resolve
    raise ResolutionImpossible(e.criterion.information)
pip._vendor.resolvelib.resolvers.ResolutionImpossible: [RequirementInformation(requirement=SpecifierRequirement('stt'), parent=None)]

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/lvyongjie/work/venv-stt/lib/python3.6/site-packages/pip/_internal/cli/base_command.py", line 189, in _main
    status = self.run(options, args)
  File "/home/lvyongjie/work/venv-stt/lib/python3.6/site-packages/pip/_internal/cli/req_command.py", line 178, in wrapper
    return func(self, options, args)
  File "/home/lvyongjie/work/venv-stt/lib/python3.6/site-packages/pip/_internal/commands/install.py", line 317, in run
    reqs, check_supported_wheels=not options.target_dir
  File "/home/lvyongjie/work/venv-stt/lib/python3.6/site-packages/pip/_internal/resolution/resolvelib/resolver.py", line 127, in resolve
    six.raise_from(error, e)
  File "<string>", line 3, in raise_from
pip._internal.exceptions.DistributionNotFound: No matching distribution found for stt
Removed build tracker: '/tmp/pip-req-tracker-1q78y9ns'

**Reply by reuben** [reply]
Weird. You're on Python 3.6 and yet it's skipping the 3.6 wheel. Are you running an old Linux distribution by any chance?

**Reply by lyjzsyzlt** [reply]
My Linux version information:
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

**Reply by reuben** [reply]
Ah, yes, that's pretty old. The manylinux_2_24 wheels require glibc >= 2.24. Upgrading glibc manually can be a risky procedure due to incompatibilities with other system packages. You can try to hack it by downloading the [3.6 wheel](https://files.pythonhosted.org/packages/1a/d8/1b80ef632896394868f7e151a5c46e95c1cf8d12db377d8ba18da33cd07a/STT-0.10.0a4-cp36-cp36m-manylinux_2_24_x86_64.whl#sha256=5f91e7c5c0a161f5e890cbf76e547e084025ef832f12916a21437f4b98965bb7) and manually renaming it to `STT-0.10.0a4-cp36-cp36m-manylinux1_x86_64.whl` and then installing the file directly with `python3 -m pip install STT-0.10.0a4-cp36-cp36m-manylinux1_x86_64.whl` but it's not guaranteed to work. Another alternative is [building from source](https://stt.readthedocs.io/en/latest/BUILDING.html).

**EDIT**: This should no longer be needed as of version 0.10.0a5. Just do a normal `python -m pip install stt`.

**Reply by lyjzsyzlt** [reply]
This works! Thanks!!!

**Reply by reuben** [reply]
FWIW I've just uploaded the new 0.10.0a5 version which should be compatible with older versions of Linux with a simple `python -m pip install stt` instead of the hack I described above.
